packages/connect/src/api/ethereum/api/ethereumSignTypedData.ts

```diff
@@ -116,10 +120,13 @@ export default class EthereumSignTypedData extends AbstractMethod<'ethereumSignT
         const cmd = this.device.getCommands();
         const { address_n, definitions } = this.params;
         if (this.device.features.internal_model === DeviceModelInternal.T1B1) {
-            validateParams(this.params, [
-                { name: 'domain_separator_hash', type: 'string', required: true },
-                { name: 'message_hash', type: 'string' },
-            ]);
+            Assert(
+                Type.Object({
+                    domain_separator_hash: Type.String(),
+                    message_hash: Type.Optional(Type.String()),
+                }),
+                this.params,
+            );
```

Comment:
I think the behaviour should be the same as before, or maybe I'm not seeing it? Before it would also throw, it just didn't assert the type for TS so it had to be typecast below